### PR TITLE
Simplify plugin code

### PR DESCRIPTION
**Pull request by ncdc**
## Summary
Simplify plugin code by adding a base Options struct

## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/cliplugins/base/options.go [hunk 1]
```diff
@@ -1,20 +1,17 @@
 /*
 Copyright 2022 The KCP Authors.
-
```
Comment (stevekuznetsov) [inline pkg/cliplugins/base/options.go:3]: What triggered this? Is it valid?

File: pkg/cliplugins/workload/cordon.go [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+		return nil
+	}
+
+	var patchBytes []byte
+	if o.Cordon {
+		patchBytes = []byte(`[{"op":"replace","path":"/spec/unschedulable","value":true}]`)
+	} else {
+		evict := ``
+		if syncTarget.Spec.EvictAfter != nil {
+			evict = `,{"op":"remove","path":"/spec/evictAfter"}`
```
Comment (stevekuznetsov) [inline pkg/cliplugins/workload/cordon.go:165]: :cry:

**Review by stevekuznetsov — COMMENTED** [review]
Seems mostly reasonable, but is very hard to review with the filepath changes.

**Comment by ncdc** [comment]
>hard to review with the filepath changes

Do you think it would be in an easier diffable state w/o the filepath changes? If so, I can try to undo that bit, then move in a separate commit

File: pkg/cliplugins/workload/cordon.go (hunk 2 above)
Comment (ncdc) [reply pkg/cliplugins/workload/cordon.go:165]: Not net new...

File: pkg/cliplugins/base/options.go (hunk 1 above)
Comment (ncdc) [reply pkg/cliplugins/base/options.go:3]: Some sort of edit/format mistake

File: pkg/cliplugins/workload/plugin/workload.go [hunk 3]
```diff
@@ -18,73 +18,83 @@ package plugin
...
 	kcpclientset "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	"github.com/kcp-dev/kcp/pkg/cliplugins/base"
 )
 
-// Cordon the sync target and mark it as unschedulable
-func (c *Config) Cordon(ctx context.Context, syncTargetName string) error {
-	config, err := clientcmd.NewDefaultClientConfig(*c.startingConfig, c.overrides).ClientConfig()
-	if err != nil {
-		return err
+// CordonOptions FIXME
```
Comment (stevekuznetsov) [inline pkg/cliplugins/workload/plugin/workload.go:33]: fix me :)

**Review by stevekuznetsov — COMMENTED** [review]
Mostly seems good! Cordon bits need a bit more

**Comment by stevekuznetsov** [comment]
Looks good but has conflicts already :(

**Comment by ncdc** [comment]
If you're good with it I'll finish it up

**Comment by stevekuznetsov** [comment]
```
 pkg/cliplugins/workspace/plugin/kubeconfig.go:650:7: ineffectual assignment to err (ineffassign)
			_, err = fmt.Fprintf(o.Out, "Updated context %q.\n", o.Name)
			   ^
pkg/cliplugins/workspace/plugin/kubeconfig.go:652:7: ineffectual assignment to err (ineffassign)
			_, err = fmt.Fprintf(o.Out, "Updated context %q and switched to it.\n", o.Name)
			   ^
pkg/cliplugins/workspace/plugin/kubeconfig.go:655:6: ineffectual assignment to err (ineffassign)
		_, err = fmt.Fprintf(o.Out, "Created context %q and switched to it.\n", o.Name)
		   ^ 
```

**Comment by ncdc** [comment]
Fixed

**Pull request merged**